ed448/src/decaf/points.rs

```diff
@@ -326,13 +326,12 @@ impl DecafPoint {
 
         let XX_TT = (X + T) * (X - T);
 
-        let (isr, _) = (X.square() * XX_TT * FieldElement::NEG_EDWARDS_D).inverse_square_root();
-        let mut ratio = isr * XX_TT;
-        let altx = ratio * FieldElement::DECAF_FACTOR; // Sign choice
-        ratio.conditional_negate(altx.is_negative());
-        let k = ratio * Z - T;
+        let (isr, _) = (X.square() * XX_TT * FieldElement::ONE_MINUS_D).inverse_square_root();
 
-        let mut s = k * FieldElement::NEG_EDWARDS_D * isr * X;
+        let mut ratio = isr * XX_TT * FieldElement::SQRT_MINUS_D;
+        ratio.conditional_negate(ratio.is_negative());
+        let k = FieldElement::INVSQRT_MINUS_D * ratio * Z - T;
+        let mut s = k * FieldElement::ONE_MINUS_D * isr * X;
```

Comment:
I don't know where the original implementation is coming from, but I [followed the specification](https://www.rfc-editor.org/rfc/rfc9496.html#section-5.3.2) here, which made things work.